src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -606,5 +615,38 @@ private ProcessWaitState GetWaitState()
         public bool Responding => true;
 
         private bool WaitForInputIdleCore(int milliseconds) => throw new InvalidOperationException(SR.InputIdleUnkownError);
+
+        private static bool s_sigchildHandlerRegistered = false;
+        private static readonly object s_sigchildGate = new object();
+        private static readonly Interop.Sys.SigChldCallback s_sigChildHandler = OnSigChild;
+
+        private static bool EnsureSigChildHandler()
+        {
+            lock (s_sigchildGate)
```

Comment:
It'd be good to do double-checked locking here, rather than having every Process.Start contend on this lock (and incur the associated locking overheads).  And rather than doing it manually, you can use `Lazy<T>`, e.g.
```C#
private static readonly Interop.Sys.SigChldCallback s_sigChildHandler = OnSigChild;
private static readonly Lazy<bool> s_sigchildRegistered = new Lazy<bool>(() =>
{
    if (Interop.Sys.InitializeSignalHandling())
    {
        Interop.Sys.RegisterForSigChld(s_sigChildHandler);
    }
    throw new Win32Exception();
});

private static bool EnsureSigChildHandler() => s_sigchildRegistered.Value;
```